x-pack/plugin/esql/build.gradle

```diff
@@ -56,7 +56,11 @@ sourceSets.main.java {
 }
 
 tasks.getByName('test') {
-  dependsOn 'cleanGeneratedDocs'
+  if (BuildParams.isCi() == false) {
+    dependsOn 'cleanGeneratedDocs'
+    finalizedBy 'copyGeneratedDocs'
```

Comment:
Why doesn't `copyGeneratedDocs` just depend on `test`? This isn't a proper use-case for `finalizedBy` because finalizer tasks are run even when the upstream task fails, which I don't think we want here.

Reply:
Ah. I wanted it to run if you run `tests`, so `dependsOn` wasn't enough.

Reply:
If that's the case then the most idomatic way would be to add a `doLast { }` to the `test` task to perform that copy as part of that task's execution.

Reply:
@mark-vieira I've swapped it. Have another look?